gtsam/base/make_shared.h

```diff
@@ -0,0 +1,44 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file make_shared.h
+ * @brief make_shared trampoline function to ensure proper alignment
+ * @author Fan Jiang
+ */
+
+#pragma once
+
+#include <boost/make_shared.hpp>
+#include <gtsam/base/types.h>
+
+#include <boost/type_traits.hpp>
+
+#include <Eigen/Core>
+
+namespace gtsam {
+
+  /**
+   * And our own `make_shared` as a layer of wrapping on `boost::make_shared`
+   */
+  template<typename T, typename ... Args>
+  boost::enable_if_t<has_custom_allocator<T>::value, boost::shared_ptr<T>> make_shared(Args&&... args)
```

Comment:
in c++11, we have enable_if no?  No need for boost or its header then...

Reply:
Since we are still on boost I think it's better to use the boost version for now, and switch when we switch all to std.

Reply:
I disagree. Let's wean ourselves from boost. I think all other uses of enable_if are already std::

Reply:
Copy